typed/measures-with-dimensions/chemistry/compound-struct.rkt

```diff
@@ -16,15 +16,17 @@ require racket/match
   ([alist : (Listof (Pairof (Un Element Compound) Natural))])
   #:transparent
   #:property prop:custom-write
-  (lambda ([comp : Compound] [out : Output-Port] [mode : (Un 0 1 #t #f)])
+  (lambda ([comp : Compound] [out : Output-Port] [mode : (Un Boolean 0 1)])
     (match mode
-      [_ #:when (andmap zero? (map cdr (compound-alist comp)))
+      [_ #:when (andmap zero? (map (lambda ([x : (Pairof (Un Element Compound) Natural)])
+                                     (cdr x))
+                                   (compound-alist comp)))
          (write-string "#<compound:>" out) (void)]
       [(or 0 #f) (display-compound comp out)]
       [(or 1 #t) (write-string "#<componud:" out)
-                 (display-compound comp out)
-                 (write-string ">" out)
-                 (void)]))
+              (display-compound comp out)
+              (write-string ">" out)
+              (void)]))
```

Comment:
Indentation should line up with the `(write-string "#<componud:" out)` on the previous line.